tiledb/api/c_api/array/test/unit_capi_array.cc

```diff
@@ -30,78 +30,1345 @@
 
 #define CATCH_CONFIG_MAIN
 #include <test/support/tdb_catch.h>
-#include "tiledb/api/c_api_test_support/testsupport_capi_context.h"
-#include "tiledb/sm/c_api/tiledb.h"
 
+#include "../array_api_experimental.h"
 #include "../array_api_external.h"
 #include "../array_api_internal.h"
+#include "tiledb/api/c_api/array_schema_evolution/array_schema_evolution_api_internal.h"
+#include "tiledb/api/c_api_test_support/testsupport_capi_array.h"
+#include "tiledb/api/c_api_test_support/testsupport_capi_context.h"
 
 using namespace tiledb::api::test_support;
 
 const char* TEST_URI = "unit_capi_array";
 
+TEST_CASE(
+    "C API: tiledb_array_schema_load argument validation", "[capi][array]") {
+  capi_return_t rc;
+  ordinary_context ctx{};
+  tiledb_array_schema_t* schema{};
+  /*
+   * No "success" section here; too much overhead to set up.
+   */
+  SECTION("null context") {
+    rc = tiledb_array_schema_load(nullptr, TEST_URI, &schema);
+    REQUIRE(tiledb_status(rc) == TILEDB_INVALID_CONTEXT);
+  }
+  SECTION("null uri") {
+    rc = tiledb_array_schema_load(ctx.context, nullptr, &schema);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null schema") {
+    rc = tiledb_array_schema_load(ctx.context, TEST_URI, nullptr);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  REQUIRE_NOTHROW(tiledb_array_schema_free(&schema));
+  CHECK(schema == nullptr);
+}
+
+TEST_CASE(
+    "C API: tiledb_array_schema_load_with_config argument validation",
+    "[capi][array]") {
+  capi_return_t rc;
+  ordinary_context ctx{};
+  tiledb_config_t* config{};
+  tiledb_array_schema_t* schema{};
+  /*
+   * No "success" section here; too much overhead to set up.
+   */
+  SECTION("null context") {
+    rc = tiledb_array_schema_load_with_config(
+        nullptr, config, TEST_URI, &schema);
+    REQUIRE(tiledb_status(rc) == TILEDB_INVALID_CONTEXT);
+  }
+  SECTION("null config") {
+    // Note: a null config is actually valid and will use the context's config.
+    // This test case merely fails without the proper overhead setup.
+    rc = tiledb_array_schema_load_with_config(
+        ctx.context, nullptr, TEST_URI, &schema);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
```

Comment:
The comment indicates that this isn't an argument validation test. It's better that it not be here; I'd just delete it and leave a comment that there's no invalid argument in this position.